File: iosApp/iosApp/Features/Definitions/Blueprints/WordDefinitionBlueprint.swift
//
//  WordDefinitionBlueprint.swift
//  iosApp
//
//  Created by Alexey Glushkov on 21.10.2020.
//  Copyright © 2020 orgName. All rights reserved.
//

import UIKit
import shared

class WordDefinitionBlueprint: Blueprint {
    typealias T = WordDefinitionViewItem
    typealias V = LabelCell
    
    var type: Int { return Int(T.Companion().Type) }

    func bind(view: V, viewItem: T) {
        view.textView.text = viewItem.firstItem() as String?
    }
}

File: androidApp/src/main/java/com/aglushkov/wordteacher/androidApp/features/definitions/blueprints/WordDefinitionBlueprint.kt
package com.aglushkov.wordteacher.androidApp.features.definitions.blueprints

import android.view.ViewGroup
import android.widget.TextView
import com.aglushkov.wordteacher.androidApp.R
import com.aglushkov.wordteacher.androidApp.features.Design
import com.aglushkov.wordteacher.androidApp.general.Blueprint
import com.aglushkov.wordteacher.androidApp.general.extensions.resolveThemeStyle
import com.aglushkov.wordteacher.androidApp.general.extensions.setTextAppearanceCompat
import com.aglushkov.wordteacher.shared.features.definitions.vm.WordDefinitionViewItem
import javax.inject.Inject

class WordDefinitionBlueprint @Inject constructor(): Blueprint<TextView, WordDefinitionViewItem> {
    override val type: Int = WordDefinitionViewItem.Type

    override fun createView(parent: ViewGroup): TextView {
        return Design.createTextView(parent).apply {
            setTextAppearanceCompat(parent.context.resolveThemeStyle(R.attr.wordDefinitionTextAppearance))
        }
    }

    override fun bind(view: TextView, viewItem: WordDefinitionViewItem) {
        view.text = viewItem.firstItem()
    }
}